deps/v8/src/heap/mark-compact.cc

```diff
@@ -2614,6 +2614,13 @@ void MarkCompactCollector::ClearWeakCollections() {
           table->RemoveEntry(i);
         }
       }
+      // Rehash if more than 25% of the entries are deleted entries.
+      // TODO(jochen): Consider to shrink the fixed array in place.
+      if ((table->NumberOfDeletedElements() << kJSWeakCollectionLoadFactorExp) >
+          table->NumberOfElements()) {
```

Comment:
The 25% doesn't seem right.  If I read the code right, it's 50%.

(Mild sense of deja vu, maybe I've commented on it before.)